apps/core/app/[locale]/(default)/(faceted)/_components/pagination.tsx

```diff
@@ -28,12 +28,18 @@ export const Pagination = ({
   const beforeSearchParams = new URLSearchParams(searchParams);
 
   beforeSearchParams.delete('after');
-  beforeSearchParams.set('before', String(startCursor));
+
+  if (startCursor) {
+    beforeSearchParams.set('before', String(startCursor));
+  }
```

Comment:
**FYI**, I've experienced issues with navigation on deleting last item on page (like Customer's address). After the action I've received url like `https://your-store/addresses?before=null` which prevents smooth navigation afterwards. So I've put this condition and if there is no cursor we just basically return start position for pagination.